### ISSUE TITLE
Changed loss function to mean_abs_rel for scale-invariant depth but training is unstable

### ISSUE THREAD

**Issue by YacineDeghaies** [prs-eth/Marigold#112]
I want the training to be more focused on relative depth so I changed the loss function in train_marigold.yaml config file to use the mean relative absolute error.

![Screenshot 2024-08-19 at 12 06 54](https://github.com/user-attachments/assets/10fab2cf-1d91-4e43-b621-59c8117fe566)

As you can see the loss is fluctuating. Any idea how to solve this ?
I even ran this for 420 epoch and the loss reaches 1.01 and stays there and never drops

I'm using these parameters:

```
optimizer:
  name: Adam

loss:
  name: mean_abs_rel
  kwargs:
    reduction: mean

lr: 3.0e-5
lr_scheduler:
  name: IterExponential
  kwargs:
    total_iter: 25000
    final_ratio: 0.01
    warmup_steps: 100
```

And this Loss function:

```

class MeanAbsRelLoss(nn.Module):
    def __init__(self, reduction='mean') -> None:
        """
        Mean Absolute Relative Loss with reduction options.

        Args:
            reduction (str, optional): Specifies the reduction to apply to the output:
                                       'none' | 'mean' | 'sum'. 'none': no reduction will be applied,
                                       'mean': the mean of the output will be taken, 'sum': the output will be summed.
        """
        super(MeanAbsRelLoss, self).__init__()
        self.reduction = reduction

    def forward(self, pred, gt):
        diff = pred - gt
        rel_abs = torch.abs(diff / gt)

        if self.reduction == 'mean':
            loss = torch.mean(rel_abs)
        elif self.reduction == 'sum':
            loss = torch.sum(rel_abs)
        elif self.reduction == 'none':
            loss = rel_abs
        else:
            raise ValueError(f"Invalid reduction mode: {self.reduction}. Supported modes: 'none', 'mean', 'sum'.")

        return loss
```

P.S : I've tried all of the loss functions in loss.py none of them work beside the mse_loss

**Comment by markkua** [comment]
Hi, the scale- and shift- invariance have been handled in the normalization step. On the other hand, the loss is applied to the latent space, not to the depth space, thus mean_abs_rel doesn't make much sense to me there.
